docs/crds/l7rule.md

```diff
@@ -0,0 +1,112 @@
+| `minPoolsUp`                         | Minimum number of UP pools to mark VS up.                                                                                                   | 0                                    |
+| `performanceLimits.maxConcurrentConnections`                         | The maximum number of concurrent client conections allowed to the Virtual Service. It is not applied on Child vs                               | Nil                                    | Nil
+| `performanceLimits.maxThroughput`         | The maximum throughput per second for all clients allowed through the client side of the Virtual Service per SE. It is not applied on Child vs                                                                                    | Nil                               |
+| `securityPolicyRef`         | Security policy applied on the traffic of the Virtual Service. This policy is used to perform security actions such as Distributed Denial of Service (DDoS) attack mitigation, etc. It is a reference to an object of type SecurityPolicy and is not applied on child vs.                                                                                       |   Nil                            |
+| `trafficCloneProfileRef`          | Server network or list of servers for cloning traffic. It is a reference to an object of type TrafficCloneProfile.                                                                                     | Nil |
+
+
+
+#### Attaching L7Rule to HostRule
+
+An L7Rule can be specifed in hostrule spec. Respective L7Rule Properties will be applied to the VS created through corresponding Hostrule. An L7Rule can be attached in the Hostrule CRD spec with `l7Rule` as the key and `name of the l7rule crd` as the value.
+
+```yaml
+apiVersion: ako.vmware.com/v1beta1
+kind: HostRule
+metadata:
+  name: my-host-rule
+spec:
+  virtualhost:
+    fqdn: test-ingclass.avi.internal     
+    fqdnType: Exact
+    l7Rule: my-l7-rule
+```
+
+
+
+#### Status Messages
+
+The status messages are used to give instantaneous feedback to the users about the reference objects specified in the L7Rule CRD.
+
+Following are some of the sample status messages:
+
+##### Accepted L7Rule object
+
+    $ kubectl get l7rule
+    NAME         STATUS     AGE
+    my-l7-rule   Accepted   3d5s
+
+
+An L7Rule is accepted only when all the reference objects specified inside it exist in the AVI Controller.
+
+##### Rejected L7Rule object
+
+    $ kubectl get l7rule
+    NAME            STATUS     AGE
+    my-l7-rule-alt  Rejected   2d23h
+    
+The detailed rejection reason can be obtained from the status:
+
+```yaml
+  status:
+    error: botPolicyRef "My-L7-Application" not found on controller
+    status: Rejected
+```
+
+#### Conditions and Caveats
+
+##### L7Rule deletion
```

Comment:
Should L7 Rule and hostrule be in the same namespace?

Reply:
What happens if hostrule deleted and not L7 Rule?

Reply:
they have to be in same namespace . 
On hostrule deletion L7Rule Properties will be also reset to default in VS

Reply:
@arihantg Lets mention: Hostrule and L7Rule should be in the same namespace.